src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/ApplicationRepository.kt

```diff
@@ -90,14 +90,15 @@ interface ApplicationRepository : JpaRepository<Application, Long>, JpaSpecifica
       "a.prisonerId = :prisonerId AND " +
       "a.modifyTimestamp >= :expiredDateAndTime AND " +
       "a.sessionSlotId = :sessionSlotId AND " +
-      "(:excludedApplicationReference is null OR a.reference != :excludedApplicationReference)",
-
+      "(:excludedApplicationReference is null OR a.reference != :excludedApplicationReference) AND " +
+      "(:excludeReservedApplicationsForUser is null OR a.createdBy != :excludeReservedApplicationsForUser)",
```

Comment:
doing it on createdBy as a a temporary measure. Raised ticket for join here - https://dsdmoj.atlassian.net/browse/VB-4231. Will do the change to do a join as part of 4231.